src/java.base/share/classes/java/nio/Direct-X-Buffer.java.template

```diff
@@ -75,13 +75,14 @@ class Direct$Type$Buffer$RW$$BO$
 
 #if[byte]
 
-    private static class Deallocator
+    private static final class Deallocator
         implements Runnable
     {
 
-        private long address;
-        private long size;
-        private int capacity;
+        private final long address;
+        private final long size;
+        private final int capacity;
+        private volatile boolean completed;
```

Comment:
Using a volatile flag here okay but something like "deallocated" might be clearer than "completed". While you there, maybe remove the spurious blank left at L81, it doesn't help readability in this tiny class.